play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/HmrcInternalHeader.scala.html

```diff
@@ -25,49 +25,53 @@
+            <path d="M67.47,7.72a59.76,59.76,0,1,0,59.75,59.75A59.75,59.75,0,0,0,67.47,7.72m67.47,59.75A67.47,67.47,0,1,1,67.47,0a67.47,67.47,0,0,1,67.47,67.47" />
           </svg>
-      }
-    <span class="hmrc-internal-header__logo-text">@language match {
+        }
+        <span class="hmrc-internal-header__logo-text">@language match {
           case Cy => { Cyllid a Thollau EF }
           case _  => { HM Revenue &amp; Customs }
         }</span>
-    </a>
-  </div>@serviceName match {
-    case Some(NonEmptyString(x)) => {<div class="hmrc-internal-header__service-name">
-        <a href="@serviceUrl" class="hmrc-internal-header__link">
-          @x
-        </a>
-      </div>}
-    case _ => {}
-  }</div>
-
+      </a>
+    </div>
+    @if(serviceName.exists(_.nonEmpty)) {
+      <div class="hmrc-internal-header__service-name">
+        @if(serviceUrl.nonEmpty) {
```

Comment:
didn't want to change the api to Option[String] because it would still have the same behaviour since empty string is falsey in js and the nunjucks doesn't check if it's undefined, it just checks if it's truthy/falsey

(though it also means that js could happily handle serviceUrl=false where as scala one wouldn't handle that)